devops/kubernetes/query-node/index.ts

```diff
@@ -4,31 +4,41 @@ import * as docker from '@pulumi/docker'
 import * as pulumi from '@pulumi/pulumi'
 import { configMapFromFile } from './configMap'
 import * as k8s from '@pulumi/kubernetes'
-import * as s3Helpers from './s3Helpers'
+import { IndexerServiceDeployment } from './indexerDeployment'
+import { ProcessorServiceDeployment } from './processorDeployment'
 import { CaddyServiceDeployment } from 'pulumi-common'
 
 require('dotenv').config()
 
 const config = new pulumi.Config()
 const awsConfig = new pulumi.Config('aws')
 const isMinikube = config.getBoolean('isMinikube')
+const externalIndexerUrl = config.get('externalIndexerUrl')
+const appsImage = config.get('appsImage') || `joystream/apps:latest`
+const skipProcessor = config.getBoolean('skipProcessor')
+const useLocalRepo = config.getBoolean('useLocalRepo') || false
+
 export let kubeconfig: pulumi.Output<any>
 export let joystreamAppsImage: pulumi.Output<string>
 let provider: k8s.Provider
 
+if (skipProcessor && externalIndexerUrl) {
+  pulumi.log.error('Need to deploy atleast one component, Indexer or Processor')
+  throw new Error(`Please check the config settings for skipProcessor and externalIndexerUrl`)
+}
+
 if (isMinikube) {
   provider = new k8s.Provider('local', {})
 
-  // Create image from local app
-  joystreamAppsImage = new docker.Image('joystream/apps', {
-    build: {
-      context: '../../../',
-      dockerfile: '../../../apps.Dockerfile',
-    },
-    imageName: 'joystream/apps:latest',
-    skipPush: true,
-  }).baseImageName
-  // joystreamAppsImage = pulumi.interpolate`joystream/apps`
+  if (useLocalRepo) {
+    // Use already existing image in minikube environment
+    joystreamAppsImage = pulumi.interpolate`${appsImage}`
+  } else {
+    // Access image from docker hub
+    joystreamAppsImage = new docker.RemoteImage('apps', {
+      name: appsImage!,
+    }).repoDigest
```

Comment:
Despite setting both isMinikube and useLocalRepo to true in the config, code path is reaching this line and I'm getting error when bringing up the stack:

```
Previewing update (query-node):
     Type                 Name                   Plan       Info
 +   pulumi:pulumi:Stack  query-node-query-node  create     1 error
 
Diagnostics:
  pulumi:pulumi:Stack (query-node-query-node):
    error: Running program '/Users/mokhtar/joystream/joystream/devops/kubernetes/query-node' failed with an unhandled exception:
    TSError: ⨯ Unable to compile TypeScript:
    index.ts(40,8): error TS2339: Property 'repoDigest' does not exist on type 'RemoteImage'.
```

Reply:
Issue with useLocalRepo not being read correctly from config solved as suggested above

```suggestion
    }).name
```
Does that work?

Reply:
`repoDigest` was added in the most recent release of the `@pulumi/docker` package. But the name works as well. I'll update the code.